hudi-common/src/main/java/org/apache/hudi/common/config/DFSPropertiesConfiguration.java

```diff
@@ -119,13 +123,13 @@ public void addPropsFromFile(Path filePath) {
     if (visitedFilePaths.contains(filePath.toString())) {
       throw new IllegalStateException("Loop detected; file " + filePath + " already referenced");
     }
-    FileSystem fileSystem;
-    try {
-      fileSystem = fs != null ? fs : filePath.getFileSystem(new Configuration());
-    } catch (IOException e) {
-      throw new IllegalArgumentException("Cannot get the file system from file path", e);
-    }
-    try (BufferedReader reader = new BufferedReader(new InputStreamReader(fileSystem.open(filePath)))) {
+
+    FileSystem fs = FSUtils.getFs(
+        filePath.toString(),
+        Option.ofNullable(hadoopConfig).orElseGet(Configuration::new)
+    );
```

Comment:
this is called just once to load props from external file right? I mean, once per write operation. looks ok to me.

Reply:
FileSystem instance is also cached at the hadoop client level. so this itself may be ok.

Reply:
Had to fix local file-system access (for properties), since it got broken after #3416 (we're passing Hadoop FS in ctor, and we're using to try to access local files)